crates/pallet-domains/src/lib.rs

```diff
@@ -1606,9 +1627,37 @@ impl<T: Config> Pallet<T> {
         Self::check_extrinsics_root(opaque_bundle)?;
 
         let proof_of_election = &sealed_header.header.proof_of_election;
+        let slot_number = proof_of_election.slot_number;
         let (operator_stake, total_domain_stake) =
             Self::fetch_operator_stake_info(domain_id, &operator_id)?;
 
+        // Check if the bundle is built with slot and valid proof-of-time that produced between the parent block
+        // and the current block
+        //
+        // NOTE: during `validate_unsigned` `current_slot` is query from the parent block and during `pre_dispatch`
+        // the `future_slot` is query from the current block.
+        if pre_dispatch {
+            ensure!(
+                slot_number <= *T::BlockSlot::future_slot(),
+                BundleError::SlotInTheFuture
+            )
+        } else {
+            ensure!(
+                slot_number > *T::BlockSlot::current_slot(),
+                BundleError::SlotInThePast
+            );
+        }
```

Comment:
Shouldn't you also check that it is not in the past in `pre_dispatch`? IIRC we should not accept bundles whose slot is at or before parent block's slot.

Reply:
This check is implicitly performed within `is_proof_of_time_valid` since we are using `frame_system::Pallet::<T>::parent_hash()`, I didn't add the check explicitly here because `pallet-domains` didn't track the `parent_slot` like `pallet-subspace`.

Reply:
Makes sense to me